tools/generate_requirements.py

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python
+"""Generate requirements/*.txt files from pyproject.toml."""
+
+import sys
+from pathlib import Path
+
+try:  # standard module since Python 3.11
+    import tomllib as toml
+except ImportError:
+    try:  # available for older Python via pip
+        import tomli as toml
+    except ImportError:
+        sys.exit("Please install `tomli` first: `pip install tomli`")
+
+repo_dir = Path(__file__).parent.parent
+
+
+def generate_requirement_file(name: str, req_list: list[str]) -> None:
+    lines = ["# Generated from pyproject.toml, do not edit"] + req_list
```

Comment:
Sorry, maybe I should have been clearer in https://github.com/scikit-image/scikit-image/pull/7085#discussion_r1284413772. I meant to include the origin of the automatically generated file. E.g.

```python
# Generated from pyproject.toml via tools/generate_requirements.py and pre-commit hook
# do not edit manually
```

Reply:
I've added a few more helpful details to the header.